src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/AutoConfirmUser/AutomaticallyConfirmOrganizationUserCommand.cs

```diff
@@ -0,0 +1,227 @@
+    }
+
+    private async Task<CommandResult<AutomaticallyConfirmOrganizationUserValidationRequest>> SendConfirmedOrganizationUserEmailAsync(
+        AutomaticallyConfirmOrganizationUserValidationRequest request)
+    {
+        try
+        {
+            var user = await userRepository.GetByIdAsync(request.OrganizationUser.UserId);
+
+            if (user is null) return new UserNotFoundError();
+
+            await mailService.SendOrganizationConfirmedEmailAsync(request.Organization.Name,
+                user.Email,
+                request.OrganizationUser.AccessSecretsManager);
+
+            return request;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Failed to send OrganizationUserConfirmed.");
+            return new FailedToSendConfirmedUserEmail();
+        }
+    }
+
+    private async Task<CommandResult<AutomaticallyConfirmOrganizationUserValidationRequest>> DeleteDeviceRegistrationAsync(
+        AutomaticallyConfirmOrganizationUserValidationRequest request)
+    {
+        try
+        {
+            var devices = (await deviceRepository.GetManyByUserIdAsync(request.OrganizationUser.UserId))
+                    .Where(d => !string.IsNullOrWhiteSpace(d.PushToken))
+                    .Select(d => d.Id.ToString());
+
+            await pushRegistrationService.DeleteUserRegistrationOrganizationAsync(devices, request.Organization.Id.ToString());
+            return request;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Failed to delete device registration.");
+            return new FailedToDeleteDeviceRegistration();
+        }
+    }
+
+    private async Task<CommandResult<AutomaticallyConfirmOrganizationUserValidationRequest>> RetrieveDataAsync(
+        AutomaticallyConfirmOrganizationUserRequest request)
+    {
+        var organizationUser = await GetOrganizationUserAsync(request);
+
+        if (organizationUser.IsError) return organizationUser.AsError;
+
+        var organization = await GetOrganizationAsync(request.OrganizationId);
+
+        if (organization.IsError) return organization.AsError;
+
+        return new AutomaticallyConfirmOrganizationUserValidationRequest
+        {
+            OrganizationUser = organizationUser.AsSuccess,
+            Organization = organization.AsSuccess,
+            PerformedBy = request.PerformedBy,
+            DefaultUserCollectionName = request.DefaultUserCollectionName,
+            PerformedOn = request.PerformedOn
+        };
+    }
+
+    private async Task<CommandResult<AcceptedOrganizationUser>> GetOrganizationUserAsync(AutomaticallyConfirmOrganizationUserRequest request)
+    {
+        var organizationUser = await organizationUserRepository.GetByIdAsync(request.OrganizationUserId);
+
+        return organizationUser switch
+        {
+            null or { UserId: null } => new UserNotFoundError(),
+            { Status: not OrganizationUserStatusType.Accepted } => new UserIsNotAccepted(),
```

Comment:
This is hiding some validation in the command even though you have a separate validator.

I suggest "user must have a userId" and "user must be in the accepted state" is part of validation logic, not fetching logic.